lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +54,117 @@ def __init__(self, path_to_config: Path) -> None:
+        return ConfigDTO(**config)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        config_dto = self._extract_config_content()
+
+        if not isinstance(config_dto.seed_urls, list):
+            raise IncorrectSeedURLError
+
+        for url in config_dto.seed_urls:
+            if not isinstance(url, str) or not re.match("https?://.*/", url):
+                raise IncorrectSeedURLError
+
+        if (not isinstance(config_dto.total_articles, int)
+            or isinstance(config_dto.total_articles, bool)
+               or config_dto.total_articles < 1):
+            raise IncorrectNumberOfArticlesError
+
+        if config_dto.total_articles > NUM_ARTICLES_UPPER_LIMIT:
+            raise NumberOfArticlesOutOfRangeError
+
+        if not isinstance(config_dto.headers, dict):
+            raise IncorrectHeadersError
+
+        if not isinstance(config_dto.encoding, str):
+            raise IncorrectEncodingError
+
+        if (not isinstance(config_dto.timeout, int) or config_dto.timeout < TIMEOUT_LOWER_LIMIT
+            or config_dto.timeout > TIMEOUT_UPPER_LIMIT):
+            raise IncorrectTimeoutError
+
+        if not isinstance(config_dto.should_verify_certificate, bool) \
+                or not isinstance(config_dto.headless_mode, bool):
+            raise IncorrectVerifyError
+
+
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self._seed_urls
 
     def get_num_articles(self) -> int:
         """
         Retrieve total number of articles to scrape
         """
-        pass
+        return self._num_articles
 
     def get_headers(self) -> dict[str, str]:
         """
         Retrieve headers to use during requesting
         """
-        pass
+        return self._headers
 
     def get_encoding(self) -> str:
         """
         Retrieve encoding to use during parsing
         """
-        pass
+        return self._encoding
 
     def get_timeout(self) -> int:
         """
         Retrieve number of seconds to wait for response
         """
-        pass
+        return self._timeout
 
     def get_verify_certificate(self) -> bool:
         """
         Retrieve whether to verify certificate
         """
-        pass
+        return self._should_verify_certificate
 
     def get_headless_mode(self) -> bool:
         """
         Retrieve whether to use headless mode
         """
-        pass
+        return self._headless_mode
 
 
 def make_request(url: str, config: Config) -> requests.models.Response:
     """
     Delivers a response from a request
     with given configuration
     """
-    pass
+    headers = config.get_headers()
+    timeout = config.get_timeout()
+    verify = config.get_verify_certificate()
+    response = requests.get(url, headers=headers, timeout=timeout, verify=verify)
+    return response
```

Comment:
consider also specifying encoding of the response here

Reply:
done